src/map/packets/currency1.cpp

```diff
@@ -47,127 +61,153 @@ CCurrencyPacket1::CCurrencyPacket1(CCharEntity* PChar)
+
+        ref<uint8>(0xCE) = static_cast<uint8>(rset->getUInt("rems_ch1"));
+        ref<uint8>(0xCF) = static_cast<uint8>(rset->getUInt("rems_ch2"));
+        ref<uint8>(0xD0) = static_cast<uint8>(rset->getUInt("rems_ch3"));
+        ref<uint8>(0xD1) = static_cast<uint8>(rset->getUInt("rems_ch4"));
+        ref<uint8>(0xD2) = static_cast<uint8>(rset->getUInt("rems_ch5"));
+        ref<uint8>(0xD3) = static_cast<uint8>(rset->getUInt("rems_ch6"));
+        ref<uint8>(0xD4) = static_cast<uint8>(rset->getUInt("rems_ch7"));
+        ref<uint8>(0xD5) = static_cast<uint8>(rset->getUInt("rems_ch8"));
+        ref<uint8>(0xD6) = static_cast<uint8>(rset->getUInt("rems_ch9"));
+        ref<uint8>(0xD7) = static_cast<uint8>(rset->getUInt("rems_ch10"));
+
+        // Salvage II Stored Plans
+        ref<uint64>(0xD8) = PackSalvageIIPlans(
+            static_cast<uint8>(rset->getUInt("bloodshed_plans")),
+            static_cast<uint8>(rset->getUInt("umbrage_plans")),
+            static_cast<uint8>(rset->getUInt("ritualistic_plans")),
+            static_cast<uint8>(rset->getUInt("tutelary_plans")),
+            static_cast<uint8>(rset->getUInt("primacy_plans")));
+
+        ref<uint16>(0xE0) = static_cast<uint16>(rset->getUInt("reclamation_marks"));
+        ref<uint32>(0xE4) = static_cast<uint32>(rset->getInt("unity_accolades"));
 
         // Crystal storage
-        ref<uint16>(0xE8) = _sql->GetUIntData(84); // Fire Crystals
-        ref<uint16>(0xEA) = _sql->GetUIntData(85); // Ice Crystals
-        ref<uint16>(0xEC) = _sql->GetUIntData(86); // Wind Crystals
-        ref<uint16>(0xEE) = _sql->GetUIntData(87); // Earth Crystals
-        ref<uint16>(0xF0) = _sql->GetUIntData(88); // Lightning Crystals
-        ref<uint16>(0xF2) = _sql->GetUIntData(89); // Water Crystals
-        ref<uint16>(0xF4) = _sql->GetUIntData(90); // Light Crystals
-        ref<uint16>(0xF6) = _sql->GetUIntData(91); // Dark Crystals
-
-        ref<uint16>(0xF8) = _sql->GetUIntData(92); // deeds
+        ref<uint16>(0xE8) = static_cast<uint16>(rset->getUInt("fire_crystals"));
+        ref<uint16>(0xEA) = static_cast<uint16>(rset->getUInt("ice_crystals"));
+        ref<uint16>(0xEC) = static_cast<uint16>(rset->getUInt("wind_crystals"));
+        ref<uint16>(0xEE) = static_cast<uint16>(rset->getUInt("earth_crystals"));
+        ref<uint16>(0xF0) = static_cast<uint16>(rset->getUInt("lightning_crystals"));
+        ref<uint16>(0xF2) = static_cast<uint16>(rset->getUInt("water_crystals"));
+        ref<uint16>(0xF4) = static_cast<uint16>(rset->getUInt("light_crystals"));
+        ref<uint16>(0xF6) = static_cast<uint16>(rset->getUInt("dark_crystals"));
+
+        ref<uint16>(0xF8) = static_cast<uint16>(rset->getUInt("deeds"));
     }
 
     ref<uint32>(0xC0) = charutils::getAvailableTraverserStones(PChar); // traverser_stones
 }
+
+uint64 CCurrencyPacket1::PackSalvageIIPlans(uint32 bloodshed, uint32 umbrage, uint32 ritualistic, uint32 tutelary, uint32 primacy)
```

Comment:
You can get rid of the packing function here with named struct initialiser syntax, I reckon

Reply:
> You can get rid of the packing function here with named struct initialiser syntax, I reckon

I think I've got that re-done correctly. The CIs weren't happy with my use of `reinterpret_cast` previously so, if it passes CIs this time, I think I'm good to squash and move forward if you like the changes.

Reply:
Nope, it griped about strict aliasing again. Would my previous commit work? I can kill the second one if so. Am open to suggestions otherwise!